backends/events/switchsdl/switchsdl-events.cpp

```diff
@@ -0,0 +1,432 @@
+/* ScummVM - Graphic Adventure Engine
```

Comment:
A lot of the code in this file appears to be duplicated from the PS Vita event manager. Can the two event managers be unified somehow?

Reply:
As one of the maintainers of the Vita port, I would rather have the Vita and Switch event managers separated if possible. I have had too much hassle in the past trying to maintain Vita and Switch combined code. The platforms are quite different after all (different numbers of buttons etc.). Even if the current code is similar, I expect that in the future we will want to make Vita and/or Switch-specific changes only to one platform.